src/gmt_remote.c

```diff
@@ -162,6 +182,19 @@ GMT_LOCAL struct GMT_DATA_INFO *gmtremote_data_load (struct GMT_CTRL *GMT, int *
 		GMT_Report (GMT->parent, GMT_MSG_ERROR, "Bad record counter in file %s\n", file);
 		return NULL;
 	}
+	if (fgets (line, GMT_LEN256, fp) == NULL) {	/* Try to get second record */
+		fclose (fp);
+		GMT_Report (GMT->parent, GMT_MSG_ERROR, "Read error second record in file %s\n", file);
+		return NULL;
+	}
+	if ((k = gmtremote_parse_version (line))) {
+		fclose (fp);
+		if (k == 2)
+			GMT_Report (GMT->parent, GMT_MSG_ERROR, "Your GMT version too old to use the remote data mechanism - please upgrade to %s or later\n", line);
+		else
+			GMT_Report (GMT->parent, GMT_MSG_ERROR, "Unable to parse %s to extract GMT version\n", line);
+		return NULL;
```

Comment:
Even though users' GMT is too old, they can still use it as long as they don't use remote dataset. Perhaps make it a notice or a warning?

Since the gmt_data_server.txt on the GMT data server is still in old format, our CI jobs report warnings like:
```
gmt [ERROR]: Unable to parse # Master table with information about all the remote data sets available on the GMT Data serve
```
Please add quotes after *parse* before *to*.

The command `gmt pscoast -R0/10/0/10 -JM6i -Ba -Ggray -ENG+p1p,blue -P -Vd > map.ps` crashes:
```
gmt [DEBUG]: Initialize FFTW with 4 threads.
gmt [DEBUG]: Local file /Users/runner/.gmt/server/gmt_data_server.txt found
gmt [DEBUG]: File /Users/runner/.gmt/server/gmt_data_server.txt less than 24 hours old, refresh is premature.
gmt [DEBUG]: Load contents from /Users/runner/.gmt/server/gmt_data_server.txt
gmt [ERROR]: Unable to parse # Master table with information about all the remote data sets available on the GMT Data server
 to extract GMT version
gmt [INFORMATION]: Unable to read server information file
gmt [DEBUG]: Local file /Users/runner/.gmt/server/gmt_hash_server.txt found
gmt [DEBUG]: File /Users/runner/.gmt/server/gmt_hash_server.txt less than 24 hours old, refresh is premature.
gmt [DEBUG]: GMT_Create_Session initialized GMT structure
gmt [DEBUG]: Loading core GMT shared library: libgmt.dylib
gmt [DEBUG]: Shared Library # 0 (core). Path = libgmt.dylib
gmt [DEBUG]: Loading GMT plugins from: /Users/runner/runners/2.170.1/work/1/s/gmt-install-dir/lib/gmt/plugins
gmt [DEBUG]: Shared Library # 1 (supplements). Path = /Users/runner/runners/2.170.1/work/1/s/gmt-install-dir/lib/gmt/plugins/supplements.so
gmt [DEBUG]: Revised options: -R0/10/0/10 -JM6i -Ba -Ggray -ENG+p1p,blue -P -Vd
pscoast [DEBUG]: History: Process -R0/10/0/10
pscoast [DEBUG]: History: Process -JM6i
pscoast [DEBUG]: History: Process -Ba
pscoast [DEBUG]: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
pscoast [DEBUG]: gmt_get_filename: In: 0/10/0/10 Out: 0/10/0/10
ERROR: Caught signal number 11 (Segmentation fault) at
0   libsystem_platform.dylib            0x00007fff7121ed45 _platform_strncmp + 325
1   ???                                 0x0000000000005825 0x0 + 22565
Stack backtrace:
0   libgmt.6.dylib                      0x00000001057da235 sig_handler + 581
1   libsystem_platform.dylib            0x00007fff712215fd _sigtramp + 29
2   ???                                 0x0000000000000001 0x0 + 1
3   libsystem_c.dylib                   0x00007fff710ce313 bsearch + 57
4   libgmt.6.dylib                      0x00000001057dd196 gmt_remote_no_extension + 134
5   libgmt.6.dylib                      0x000000010584b415 gmt_access + 421
6   libgmt.6.dylib                      0x00000001058ef889 gmt_parse_R_option + 1481
7   libgmt.6.dylib                      0x0000000105917311 gmt_parse_common_options + 6897
8   libgmt.6.dylib                      0x00000001058df13b GMT_Parse_Common + 5067
```

Reply:
Did 1 and 2.  3 dies not crash for me.  I suppose it will if not updating the remote file soon

Reply:
If you remove your local gmt_data_server.txt, you may see the crash.

Reply:
Sure, but once we upload the new file...